src/dmd/lib.d

```diff
@@ -95,10 +95,19 @@ class Library
         if (global.params.verbose)
             message("library   %s", loc.filename);
 
-        OutBuffer libbuf;
+        import core.stdc.string : strlen;
+        auto tmpname = loc.filename[0 .. loc.filename.strlen] ~ ".tmp\0";
+        scope(exit) destroy(tmpname);
+
+        auto libbuf = OutBuffer(tmpname.ptr);
```

Comment:
Here the code converts a 0 terminated string to a D array, then uses it as a 0 terminated string. In line 104, it converts it again to a D array, then in lines 106 and 108 uses it as a 0 terminated string again.

Reply:
6d6cd8fc7a3de791662958ea46be0005c6e5dc85 improves on that a bit.